simapp/v2/app_di.go

```diff
@@ -74,127 +83,124 @@ func AppConfig() depinject.Config {
+			//
+			// func() runtime.ValidatorAddressCodec { return <- custom validator address codec type -> }
+			// func() runtime.ConsensusAddressCodec { return <- custom consensus address codec type -> }
+
+			//
+			// MINT
+			//
+
+			// For providing a custom inflation function for x/mint add here your
+			// custom function that implements the minttypes.InflationCalculationFn
+			// interface.
 			),
 			depinject.Provide(
-				// inject desired account types:
-				multisigdepinject.ProvideAccount,
-				basedepinject.ProvideAccount,
-				lockupdepinject.ProvideAllLockupAccounts,
-
-				// provide base account options
-				basedepinject.ProvideSecp256K1PubKey,
-				// if you want to provide a custom public key you
-				// can do it from here.
-				// Example:
-				// 		basedepinject.ProvideCustomPubkey[Ed25519PublicKey]()
-				//
-				// You can also provide a custom public key with a custom validation function:
-				//
-				// 		basedepinject.ProvideCustomPubKeyAndValidationFunc(func(pub Ed25519PublicKey) error {
-				//			if len(pub.Key) != 64 {
-				//				return fmt.Errorf("invalid pub key size")
-				//			}
-				// 		})
+			// if you want to provide a custom public key you
+			// can do it from here.
+			// Example:
+			// 		basedepinject.ProvideCustomPubkey[Ed25519PublicKey]()
+			//
+			// You can also provide a custom public key with a custom validation function:
+			//
+			// 		basedepinject.ProvideCustomPubKeyAndValidationFunc(func(pub Ed25519PublicKey) error {
+			//			if len(pub.Key) != 64 {
+			//				return fmt.Errorf("invalid pub key size")
+			//			}
+			// 		})
 			),
 		)
 	)
 
-	if err := depinject.Inject(appConfig,
+	outputs = append(outputs,
+		&logger,
 		&storeBuilder,
 		&appBuilder,
 		&app.appCodec,
 		&app.legacyAmino,
 		&app.txConfig,
 		&app.interfaceRegistry,
 		&app.UpgradeKeeper,
-		&app.StakingKeeper,
-	); err != nil {
-		panic(err)
-	}
+		&app.StakingKeeper)
 
-	// store/v2 follows a slightly more eager config life cycle than server components
-	storeConfig, err := serverstore.UnmarshalConfig(viper.AllSettings())
-	if err != nil {
-		panic(err)
+	if err := depinject.Inject(appConfig, outputs...); err != nil {
+		return nil, err
 	}
 
-	app.store, err = storeBuilder.Build(logger, storeConfig)
+	var err error
+	app.App, err = appBuilder.Build()
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
-	app.App, err = appBuilder.Build()
-	if err != nil {
-		panic(err)
+	app.store = storeBuilder.Get()
+	if app.store == nil {
+		return nil, fmt.Errorf("store builder not return a db")
 	}
 
 	/****  Module Options ****/
 
 	// RegisterUpgradeHandlers is used for registering any on-chain upgrades.
 	app.RegisterUpgradeHandlers()
 
-	// TODO (here or in runtime/v2)
-	// wire simulation manager
-	// wire unordered tx manager
+	if err = app.LoadLatest(); err != nil {
+		return nil, err
+	}
+	return app, nil
+}
 
-	if err := app.LoadLatest(); err != nil {
+// NewSimAppWithInputs returns a reference to an initialized SimApp.
```

Comment:
I see it used in tests only, can we kill?

Reply:
the tests which use this are in the same `simapp` package, so idt it can be killed, maybe moved? where do you think it should go?

Reply:
Just not have an helper and use NewSimapp in that test?

Reply:
makes sense 👍